### PR TITLE
EN-2343: Show message when credentials are not provided

### PR DESCRIPTION
**Pull request by JonathanLoscalzo**

### CODE REVIEW

File: iambic/config/wizard.py [hunk 1]
```diff
@@ -510,15 +512,21 @@ def secret_config_dir(self, secrets_file: str = "secrets.yaml") -> str:
...
+        self.config_path = await self._get_config_path()
 
         if os.path.exists(self.config_path) and os.path.getsize(self.config_path) != 0:
             log.info("Found existing configuration file", config_path=self.config_path)
             try:
                 self.config = await load_config(self.config_path)
+            except NoCredentialsError:
+                question_text = (
+                    "We couldn't find your AWS credentials. "
+                    "Please estart the wizard with valid AWS credentials "
```
Comment (castrapel) [inline iambic/config/wizard.py:524]: Could you use the same message and log status (error vs info) as we use here? https://github.com/noqdev/iambic/blob/main/iambic/plugins/v0_1_0/aws/utils.py#L219

File: iambic/plugins/v0_1_0/aws/utils.py [hunk 2]
```diff
@@ -220,7 +219,7 @@ def get_current_role_arn(sts_client) -> str:
             "Unable to detect AWS Credentials. Please follow the guide here "
             "to set up AWS credentials: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html"
         )
-        sys.exit(1)
+        raise
```
Comment (castrapel) [inline iambic/plugins/v0_1_0/aws/utils.py:222]: Just saw this one - why raise here, instead of just exiting?

Comment (JonathanLoscalzo) [reply iambic/plugins/v0_1_0/aws/utils.py:222]: I guess that the "wizard" is responsible of stop the application, and not the plugin

**Review by smoy — COMMENTED** [review]
I put this as non-blocking comment since the code does work as it is.

The @functools.wrap is a little odd because we simply decorate the original version with some try-catch capability and name it exactly the same. I would ask why not just modify the original `iambic.config.dynamic_config::load_config` implementation.

Now from reading `iambic.config.dynamic_config::load_config` implementation, we haven't designed how the various co-operative pattern on the plugins. For example: it's a single config file including aws, okta, google_workspace config. In this particular JIRA issue, the AWS configure_plugin should define the pattern it used when encounters try-catch boto3.NoCredentialsError and re-raise a iambic-specific exception with the message, "hi user, you are using the aws provider and it requires to have AWS_PROFILE or the typical AWS specific environment variable setup". If we don't design a provider neutral methodology, ``iambic.config.dynamic_config` will have to import a lot of provider specific packages (circular import headache and what not).

**Comment by JonathanLoscalzo** [comment]
> I put this as non-blocking comment since the code does work as it is.
> 
> The @functools.wrap is a little odd because we simply decorate the original version with some try-catch capability and name it exactly the same. I would ask why not just modify the original `iambic.config.dynamic_config::load_config` implementation.
> 

Indeed, there are other places where the load_config is being invoke, and I don't know if it should have that except's. For Example, in github.py. In those cases, we should catch that errors? or are just part of the wizard? I mean, I'd prefer to raise an exception in a plugin rather than execute a sys.exit(N).

> Now from reading iambic.config.dynamic_config::load_config implementation, we haven't designed how the various co-operative pattern on the plugins. For example: it's a single config file including aws, okta, google_workspace config. In this particular JIRA issue, the AWS configure_plugin should define the pattern it used when encounters try-catch boto3.NoCredentialsError and re-raise a iambic-specific exception with the message, "hi user, you are using the aws provider and it requires to have AWS_PROFILE or the typical AWS specific environment variable setup". If we don't design a provider neutral methodology, ``iambic.config.dynamic_config` will have to import a lot of provider specific packages (circular import headache and what not).

Yes, we could manage named exception on each plugin (custom exceptions), but the manager (in this case, the wizard) should wrap those exceptions as it expects depending on the case. 

Another option could have generic plugins exception, one for each plugin function (LoadException, DetectException, and so on), and the plugins should have the responsibility of raise the correct ones.

**Review by smoy — APPROVED** [review]
works for me locally.

```
ugpy/launcher 65534 -- -m iambic.main plan 
2023/08/04 14:55:23 [info     ] Loading config...
2023/08/04 14:55:23 [info     ] Setting config metadata...
2023/08/04 14:55:25 [error    ] Unable to detect AWS Credentials. Please follow the guide here to set up AWS credentials: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
2
```

**Pull request merged**